packages/samples/headless-react/src/pages/SearchPage.tsx

```diff
@@ -290,6 +304,10 @@ export class SearchPage extends Component {
   }
 
   private executeInitialSearch() {
+    if (this.engine.state.search.response.searchUid) {
```

Comment:
I think we could also use the search status controller for this, would be cleaner/more straightforward why that check exists.